conans/client/build/autotools_environment.py

```diff
@@ -57,9 +57,17 @@ def __init__(self, conanfile, win_bash=False, include_rpath_flags=False):
         # Not declared by default
         self.fpic = None
 
-    def _get_triplet(self, the_arch, the_os):
+        # Precalculate build, host, target triplets
+        self.build, self.host, self.target = self._get_host_build_target_flags()
+
+    @staticmethod
+    def get_triplet(conanfile, the_arch, the_os):
```

Comment:
I didn't remember that the conanfile was being used in this method, feels a little weird being a static method but passing a conanfile when the instance already has it.

As a triplet is a quite generic concept, related with the building, of course, but not pure building concept, should we extract the triplet function to a tool with an optional `compiler` parameter and just raising an exception when it is not able to generate a triplet (to avoid the output)? @memsharded WDYT?

Reply:
Agree having the conanfile as parameter is weird. We could simple pass the compiler (needed to calculate the triplet at some point) and change the warning output to an exception. For me having this in tools instead of in the build helper seems smoother

Reply:
I would make the compiler optional, it is only needed in the case of windows (where it is not common at all to use it), so maybe we could raise in windows if not a compiler and let it optional by default.